### DISCUSSION TITLE
Last call for issues for Regolith 3.0

### DISCUSSION THREAD

**Discussion by kgilmer** [regolith-linux/regolith-desktop#876] [Announcements]
Regolith Desktop 3.0 has been in beta for several weeks now.  This thread is to mention any existing issues or problems directly with Regolith 3.0 beta 2 that are serious enough that they must be addressed before the release can proceed.  The Wayland session will continue to be in "beta" status for this release. If nothing serious and definitive is reported, the final release of 3.0 will proceed.

**Comment by mvsrgc** [comment]
I'm not sure if this is the correct place to post this but, here are issues I experience right now:

Regolith 3 Beta 2 on Wayland issues:

- Super + ? does not bring up the keybindings menu anymore, is this normal? (`ilia keybindings` command opens the Super + Space menu with only Applications and Help tabs.
- The Logout program from Super + Space does nothing but Super + Escape works correctly.
- I set wm.gaps.inner.size: 10 in ~/.config/regolith3/Xresources and it has no effect. (`trawlcat wm.gaps.inner.size` outputs `10`.

Regolith 3 Beta 2 on X11:
- Super + ? brings up the keybindings menu but it is empty, the Files tab in that menu is also empty. (`ilia keybindings` command behaves like on Wayland, it opens the Applications menu.
- wm.gaps.innser.size: 10 in ~/.config/regolith3/Xresources has no effect (`xrescat wm.gaps.inner.size` outputs `10`)

Super + space and Super + n, and workspace changes are working perfectly fine on Wayland and X11.

Thanks!

**Reply by kgilmer** [reply]
That's a huge help @mvsrgc , thank you!  if you'd be interested in some regolith laptop stickers send me a mailing address to kgilmer at gmail and I'll send you some.

**Reply by Vosjedev** [reply]
I am also experiencing some issues with the `~/.config/regolith/Xresources` file, see [https://github.com/orgs/regolith-linux/discussions/889](https://github.com/orgs/regolith-linux/discussions/889).

**Comment by falko** [comment]
On Wayland my screen settings, e.g. scaling to 200% and position of screens, are lost after every reboot.

**Reply by kgilmer** [reply]
@SoumyaRanjanPatnaik is this a known issue or regression?

**Reply by SoumyaRanjanPatnaik** [reply]
It's not a known issue. I'll take a look

**Reply by SoumyaRanjanPatnaik** [reply]
I think I know what's going on here. I think this is a naming issue.

**Reply by SoumyaRanjanPatnaik** [reply]
I don't immediately have access to a secondary monitor. @falko can I get the output for the following? 
```
cat ~/.config/regolith3/kanshi/profiles/*
```
```
cat ~/.config/regolith3/kanshi/config
```
and 
```
cat ~/.config/regolith3/kanshi/profiles/*
```

**Reply by falko** [reply]
I only have a `kanshi` folder in `~/.config/regolith2` but not in `~/.config/regolith3`. Could that be the issue?

**Reply by falko** [reply]
I symlinked `~/.config/regolith2/kanshi` into `~/.config/regolith3`and now it starts with the saved settings.

**Comment by jc00ke** [comment]
Which control center should we have installed? Looks like I have `gnome-control-center`, `regolith-control-center` and `regolith-sway-control-center` all installed, and none of them start when invoked from `ilia`.

**Reply by SoumyaRanjanPatnaik** [reply]
Good find. I did discover this issue earlier but almost forgot about it lol.

**Reply by kgilmer** [reply]
really?  where is `regolith-sway-control-center` coming from?

**Reply by SoumyaRanjanPatnaik** [reply]
There's no regolith-sway-control-center. But I know what the issue is.

**Reply by SoumyaRanjanPatnaik** [reply]
The issue is not `regolith-control-center` not starting from ilia, but rather `ilia` not showing `regolith-control-center` in the application list. This is an issue that  I had discovered earlier but forgot about it since I typically access `regolith-control-center` using `Super+c`.

**Reply by SoumyaRanjanPatnaik** [reply]
This happens because the `XDG_CURRENT_DESKTOP` env variable is not set to something that contains `GNOME`. This doesn't happen in `regoliith-session-flashback` because of this line in `/usr/bin/regolith-session`
```
 export XDG_CURRENT_DESKTOP="Regolith:GNOME-Flashback:GNOME"
```

**Reply by SoumyaRanjanPatnaik** [reply]
This happens because of `OnlyShowIn=GNOME;` line in the desktop file for gnome-control-center.

**Reply by SoumyaRanjanPatnaik** [reply]
Interestingly, I don't see a desktop file for regolith-control-center.

**Reply by kgilmer** [reply]
Hmm, yeah

```
~/dev/repos/gnome-control-center$ find . -name \*.desktop.in
...
./shell/org.gnome.Settings.desktop.in

~/dev/repos/regolith-control-center$ find . -name \*.desktop.in
./tests/interactive-panels/applications/gtp-toplevel-sidebar-widget.desktop.in
./tests/interactive-panels/applications/gtp-header-widget.desktop.in
./tests/interactive-panels/applications/gtp-sidebar-widget.desktop.in
./tests/interactive-panels/applications/gtp-static-init.desktop.in
./tests/interactive-panels/applications/gtp-dynamic-panel.desktop.in
```

This looks like a regression from my porting of gnome-control-center.  I'll take this issue unless you're already in flight @SoumyaRanjanPatnaik , LMK

**Reply by SoumyaRanjanPatnaik** [reply]
yeah, you can go ahead. I'll  probably also be doing a small change in how ilia is launched to deal with the `OnlyShowIn=GNOME` line of the desktop file for gnome-control-center

**Reply by kgilmer** [reply]
still digging but seems to be a problem specific to `lunar`, or at least `jammy` is shipping the desktop files.

**Reply by kgilmer** [reply]
the control center desktop file issue on lunar is fixed in `unstable`.  this was also apparently present in the last release (2.2), but unreported.

**Comment by jc00ke** [comment]
On my desktop running under Wayland, with no battery, the `battery` block of `i3status-rust` reports an empty battery. It doesn't seem to pick up that it's missing, even though I have `missing_format` specified:

```toml
[[block]]
block = "battery"
format = " $icon\u2009 $percentage "
missing_format = ""
device = "displaydevice"
driver = "upower"
```

**Reply by SoumyaRanjanPatnaik** [reply]
Although this is an issue with i3status-rs, I think there must be a properly documented solution for this. Regardless, this is definitely relevant within the context of this discussion as we cannot assume all `regolith-wayland` users to have a laptop that runs on battery. I'll dig into the docs and try to find out what's going on.

**Reply by kgilmer** [reply]
wouldn't this just be a matter of the i3status-rs config file having a battery block only in cases that the machine has a battery?  were you thinking that i3status-rs should be more intelligent about hw detection @SoumyaRanjanPatnaik ?

**Reply by SoumyaRanjanPatnaik** [reply]
Yes. That's what I was thinking. I think `i3status-rs` might already have this.

**Comment by theova** [comment]
Regolith seems to use only the first input source listed in the Ubuntu settings.

This is probably a bug.

Related discussions:
* https://github.com/orgs/regolith-linux/discussions/846#discussioncomment-5719714
* https://github.com/orgs/regolith-linux/discussions/878

**Reply by kgilmer** [reply]
Input source management I believe is handled by `gnome-flashback` for the X session.  This isn't something easily resolved for the 3.0 release.

**Reply by SoumyaRanjanPatnaik** [reply]
We can add a keybinding for this. Maybe also create a page for ilia that does this

**Reply by kgilmer** [reply]
@SoumyaRanjanPatnaik can you write up more details of your ideas in an issue?

**Comment by cjthompson** [comment]
Are there duplicate packages?

I have `regolith-wm-ilia` and `regolith-i3-ilia` packages installed and they conflict causing an i3 config error.

**Reply by kgilmer** [reply]
Yep, that's a bug, thanks @cjthompson !

**Reply by kgilmer** [reply]
this should be resolved in `unstable`

**Comment by kgilmer** [comment]
some new issues reported in slack:

1:

```
I just upgraded a laptop to Ubuntu Lunar and Regolith 3.0 to test and it seems that the autostart with wm.program.(1…3) is not working anymore.

I’m using the i3-alternating-layout script and it doesn’t work
```

2:

```
Found a bug with regolith 3.0 dependency : you can install the whole thing without i3-wm and have it crash
I found it because I still add the old i3-wm-gaps. So I removed it then I was not able to login anymore
```

Will investigate before rolling the next beta.  @SoumyaRanjanPatnaik LMK if you have any work in flight that we should wait for.

**Reply by kgilmer** [reply]
asking for clarification for the first item (xres key rename), made fix for second issue in [`regolith-wm-config`](https://github.com/regolith-linux/regolith-wm-config/commit/3b6d123bcf7c24fd7259bfce0de12830332fd6d4).

**Reply by georglauterbach** [reply]
@kgilmer I am seeing this issue as well: just upgraded to 3.0 and autostart (using `wm.program.N`) is not working anymore.

**Comment by jc00ke** [comment]
What are you all using for screenshots and marking up those shots? I've used flameshot for years, but it's not copying the shot to the clipboard. I found some help docs that make me think it would if the ENV vars are set correctly for flameshot (I haven't tried yet) but they might mess with Regolith ENV vars. For instance:

```
> echo $XDG_DESKTOP_SESSION
regolith-sway
```

According to [the docs](https://github.com/flameshot-org/flameshot/blob/master/docs/Sway%20and%20wlroots%20support.md#basic-steps) that ENV var needs to be just `sway`.

**Comment by cfsmp3** [comment]
I use flameshot, but indeed copy to clipboard is broken (however saving to
file is not).

There's an open issue for it:
https://github.com/flameshot-org/flameshot/issues/3142

**Reply by jc00ke** [reply]
Thanks, I've subscribed to that issue.

**Comment by kgilmer** [comment]
https://github.com/regolith-linux/regolith-desktop-website/pull/28 is available for review

**Reply by SoumyaRanjanPatnaik** [reply]
Looks good to me.

**Comment by kgilmer** [comment]
Regolith 3.0 Beta 3 has been released to `testing`: https://regolith-desktop.com/docs/reference/Releases/regolith-3.0-release-notes/

**Comment by jc00ke** [comment]
How might one configure/disable `swayidle`?

**Reply by kgilmer** [reply]
Did you see this Jesse? https://github.com/regolith-linux/trawl/blob/master/examples/partials.d/swayidle

**Reply by jc00ke** [reply]
I hadn't, I'll take a look, thanks!

**Reply by jc00ke** [reply]
I'm not seeing where I would make those changes. I'm not familiar with `trawl`. Would I make these changes in my `$HOME/.config/regolith3/Xresources` file or somewhere else?

**Reply by SoumyaRanjanPatnaik** [reply]
The trawl method is no longer relevant. `swayidle` is now completely controlled by `regolith-control-center` / `gnome-control-center`. You can adjust its settings in the `Power` panel of the settings apps.

**Reply by jc00ke** [reply]
Thanks, I'll check in settings. Speaking of, which one should I have installed? I have both, but `gnome-control-center` crashes on startup:

```
➜ gnome-control-center
13:26:23.9582             GLib-GIO[21880]:   ERROR: Settings schema 'org.gnome.shell.app-switcher' is not installed
fish: Job 1, 'gnome-control-center' terminated by signal SIGTRAP (Trace or breakpoint trap)
```

And nothing happens when I try to open `Settings` from `ilia`, maybe it's trying to start `gnome-control-center`?

**Reply by jc00ke** [reply]
Yep, when I changed `gnome-control-center` to `regolith-control-center` in `/usr/share/applications/org.gnome.Settings.desktop` then the settings app opens.

**Reply by kgilmer** [reply]
Hmm, do you not have another "desktop" file for `regolith-control-center`?  

```
$ dpkg -S /usr/share/applications/org.regolith.Settings.desktop
regolith-control-center: /usr/share/applications/org.regolith.Settings.desktop
```

**Reply by jc00ke** [reply]
I do have that, yes. I feel like the last time I tried uninstalling `gnome-control-center` there was an issue with dependencies and `regolith-control-center` wouldn't start. It was a while ago. Should I uninstall `gnome-control-center`?

**Reply by kgilmer** [reply]
No, in fact `regolith-control-center` is mostly a subset of the gnome version, with some tweaks for i3 and modal pop-ups, and depends on the gnome variant for some UI resources.  They are intended to be installed together in a Regolith system.

**Reply by jc00ke** [reply]
Cool, that makes sense. So when I want to open it with `ilia`, I'll make sure to select the `Regolith Settings` one. I think my eyes never scanned to the 2nd result.

**Reply by jc00ke** [reply]
@SoumyaRanjanPatnaik I have automatic suspend disabled yet my office desktop just suspended. Is there a setting I should expect to change when I flip the value in the UI?

![2023-08-14_09-42](https://github.com/regolith-linux/regolith-desktop/assets/18191/1b5389e1-dee6-4f97-a8f7-f3a5e0290549)

**Reply by jc00ke** [reply]
@SoumyaRanjanPatnaik according to the screenshot above, would you expect the computer to suspend automatically? Do you know of a different setting I should set to prevent automatic suspend?

I ask because yesterday I woke my computer up after an automatic suspend and I couldn't type in the `swaylock` password field. I could move my mouse but my keyboard didn't work. I had to reboot to get it working again.

**Reply by SoumyaRanjanPatnaik** [reply]
I just checked out the code. I'm pretty sure it shouldn't suspend. I wonder whats wrong.

**Reply by jc00ke** [reply]
@kgilmer any more guidance you could give me on how to set `swayidle`? Would I set it in my `Xresources` file with the `swayidle.*` values you referenced in https://github.com/regolith-linux/trawl/blob/master/examples/partials.d/swayidle ?

So far using Wayland with sway has been great, outside of this issue (which, personally, I'd consider a blocker for 3.0 given it effectively turns my computer off when I don't want it turned off :wink: )

**Reply by pmahnke** [reply]
I cannot seem to run regolith-control-center on mantic.

```regolith-control-center                                      

** (regolith-control-center:229427): WARNING **: 13:31:48.485: Ignoring broken panel diagnostics (missing desktop file)

** (regolith-control-center:229427): WARNING **: 13:31:48.486: Ignoring broken panel location (missing desktop file)

** (regolith-control-center:229427): WARNING **: 13:31:48.487: Ignoring broken panel thunderbolt (missing desktop file)

** (regolith-control-center:229427): WARNING **: 13:31:48.487: Ignoring broken panel usage (missing desktop file)
**
ERROR:../shell/cc-shell-model.c:419:cc_shell_model_set_panel_visibility: assertion failed: (valid)
Bail out! ERROR:../shell/cc-shell-model.c:419:cc_shell_model_set_panel_visibility: assertion failed: (valid)
[1]    229427 IOT instruction (core dumped)  regolith-control-center```

**Reply by austinweisgrau** [reply]
Also seeking guidance here. I would like to be able to have full configurability over swayidle outside the limited options in the regolith control center UI, but not seeing where that can be configured. Attempting to override swayidle parameters in XResources does not seem to work.

**Reply by kgilmer** [reply]
@SoumyaRanjanPatnaik any suggestions or links to docs for @austinweisgrau ?

**Reply by kgilmer** [reply]
@pmahnke should be working now.

**Comment by kgilmer** [comment]
Has anyone had a chance to take beta 3 for a spin yet?  Any additional regressions or not seeing expected fixes?

**Reply by jc00ke** [reply]
I'm not able to reboot from `ilia`.

**Reply by jc00ke** [reply]
Shutdown works :laughing:

**Reply by kgilmer** [reply]
I notice I have to run reboot *twice* on the sway session for it to work, but on X11 both actions work as expected.

**Reply by SoumyaRanjanPatnaik** [reply]
I don't see options for both. I assume there are packages for this that the X11 session installs but the sway session doesn't

**Reply by kgilmer** [reply]
Well the sway session is getting these actions because they are shipping in package `regolith-default-settings`, and that is a hard dependency from `regolith-desktop`.  The commands these actions are executing behind the scenes are gnome-specific so it's surprising they even work a bit.  After reviewing the contents of the `regolith-default-settings` package I'm wondering if any of this is useful for sway.  I'm currently [building a change into `unstable`](https://github.com/regolith-linux/voulage/actions/runs/5849248279) that pushes the dependency on `regolith-default-settings` from `regolith-desktop` down to `regolith-session-flashback` to see what shakes out.

**Reply by jc00ke** [reply]
> I notice I have to run reboot twice on the sway session for it to work

Same here, just tested on my laptop.

**Reply by kgilmer** [reply]
I didn't notice any issues w/ a fresh wayland session  install on Lunar without `regolith-default-settings`.  @SoumyaRanjanPatnaik LMK if you can think of any issues w/ not having this package installed.

**Reply by SoumyaRanjanPatnaik** [reply]
For some reason, I don't seem to have that package, and I never even noticed it was absent.

**Reply by SoumyaRanjanPatnaik** [reply]
Do I don't expect any new issues to come up due to this

**Reply by kgilmer** [reply]
hmm, weird.  @SoumyaRanjanPatnaik do you have the `regolith-desktop` package installed?  if so what version?

**Reply by SoumyaRanjanPatnaik** [reply]
I didn't have it installed.

**Reply by SoumyaRanjanPatnaik** [reply]
Is it not a hard dependency?

**Reply by SoumyaRanjanPatnaik** [reply]
Btw I think I figured this out. `gnome-session-quit` is supposed to open a dialog for all the : `Logout`, `Reboot`and `Shutdown`. But the gui for that is probably written in a version of GTK that doesn't support sway. Since we're still using `gnome-session` to start the sway session, I had to use the `gnome-session-quit` with the `--no-prompt` flag. This means we skip the gui, but also, logout, reboot, etc happen as soon as the command is executed without confirmation from the user

**Comment by jc00ke** [comment]
One thing I've seen that's been kinda tricky to repro is `i3status-rust` bar will crash if I double click an icon. I know it's crashed when I double clicked the Zoom icon, and today it crashed when double clicking the spinning network icon (when it was trying to connect after suspending).

How might I go about debugging it? I have to reload `sway` to get the bar to start up again, or at least that's what I've done so far.

**Reply by SoumyaRanjanPatnaik** [reply]
I can reproduce this. Actually `swaybar` doesn't really support interactions with tray icons. The way I got them to work was to apply the patch for the WIP PR (which was based on `sway v1.8`) on top of `sway v1.7` with a few changes to make everything work. The best I can do now is to take the current version of the PR (which has been rebased on `v1.9`).

**Comment by jc00ke** [comment]
This probably shouldn't hold up any potential releases, but I'm seeing a confusing nag warning:

> [!WARNING]  
> Warning on line 9 (/usr/share/regolith/sway/config.d/35_gaps) 'bindsym $mod+$wm.binding.inc_gaps gaps inner current plus 6': Overwriting binding 'Mod4+plus' for device '*' to `gaps inner current plus 6` from `scratchpad show`

```
###############################################################################
# gaps Window Resize
###############################################################################

## Resize // Window Gaps // <> + - ##
set_from_resource $wm.binding.inc_gaps wm.binding.inc_gaps plus
set_from_resource $wm.binding.dec_gaps wm.binding.dec_gaps minus
bindsym $mod+$wm.binding.dec_gaps gaps inner current minus 6
bindsym $mod+$wm.binding.inc_gaps gaps inner current plus 6   <--- this line
```
```
# from my overrides
bindsym --no-warn $mod+plus exec "swaymsg scratchpad show"
```

I thought the `--no-warn` option on `bindsym` would do the trick. Ideas? Thanks!

**Reply by jc00ke** [reply]
If I comment out line 9, then I get an error on line 14:

> [!WARNING]  
> Warning on line 14 (/usr/share/regolith/sway/config.d/35_gaps) 'bindsym --to-code $mod+$wm.binding.dec_gaps gaps inner current minus 12': Unknown key or button '"$alt'

When I comment out lines 14 and 15 then sway will reload w/o errors. Here's the file in its current state:

```
###############################################################################
# gaps Window Resize
###############################################################################

## Resize // Window Gaps // <> + - ##
set_from_resource $wm.binding.inc_gaps wm.binding.inc_gaps plus
set_from_resource $wm.binding.dec_gaps wm.binding.dec_gaps minus
#bindsym $mod+$wm.binding.dec_gaps gaps inner current minus 6
#bindsym $mod+$wm.binding.inc_gaps gaps inner current plus 6

## Resize // Big Window Gaps // <><Shift> + - ##
set_from_resource $wm.binding.inc_gaps_big wm.binding.inc_gaps_big Shift+plus
set_from_resource $wm.binding.dec_gaps_big wm.binding.dec_gaps_big Shift+minus
#bindsym --to-code $mod+$wm.binding.dec_gaps_big gaps inner current minus 12
#bindsym --to-code $mod+$wm.binding.inc_gaps_big gaps inner current plus 12

###############################################################################
# gaps Appearance and Behavior
###############################################################################

# Specify the distance between windows in pixels
set_from_resource $wm.gaps.inner.size wm.gaps.inner.size 5
set_from_resource $wm.gaps.outer.size wm.gaps.outer.size 0
gaps inner $wm.gaps.inner.size
gaps outer $wm.gaps.outer.size

# Only enable gaps on a workspace when there is at least one container
set_from_resource $wm.gaps.smart wm.gaps.smart on
smart_gaps $wm.gaps.smart
```

**Comment by kgilmer** [comment]
Despite the late hour of this release, I decided to take a few tihngs in:
* fix idle monitor on x11 
* allow a look to be installed and loaded directly on first session load without having to manually select it (x11, wayland)
* default to installing the picom compositor on x11

I'm going to change the website for the 3.0 release to move insall info to a dedicated page.  The tabs per OS version trip people up.  And given that we now have two sessions, I think specifying the desktop (meta) package, a session package (wayland/x11), and a look.

**Comment by jc00ke** [comment]
https://github.com/regolith-linux/regolith-desktop/assets/18191/c3dde135-4b07-4f05-afd9-41183e509367


I've been seeing cursor size changing lately when moving from one window to another (though not all windows). Anyone have an idea on what's going on, or how to prevent it?

**Reply by kgilmer** [reply]
I have noticed some mouse cursor weirdness as well (cursor disappears in some apps) but haven't pinned it down on anything yet.  Seems specific to the wayland session.

**Reply by jc00ke** [reply]
I've seen the cursor disappear too.

**Reply by SoumyaRanjanPatnaik** [reply]
In wayland, each application is responsible for setting its own cursor. In most cases they take up the system cursor, but that might not always be the case. Bugs in an application may be the cause of disappearing cursors

**Comment by Vosjedev** [comment]
~~I am experiencing some issues with the `~/.config/regolith/Xresources` file, see [https://github.com/orgs/regolith-linux/discussions/889](https://github.com/orgs/regolith-linux/discussions/889).~~

Also, notifications don't work for me, I can send as much `notify-send`s as i want, none of them appear...

**Comment by kgilmer** [comment]
Regolith 3.0 Beta 4 has been released to testing: https://regolith-desktop.com/docs/reference/Releases/regolith-3.0-release-notes/

**Comment by digital-tinta** [comment]
Even though it's not a "coding issue", all of the sudden I started receiving this updating error:

"Err:7 https://regolith-desktop.org/testing-ubuntu-lunar-amd64 lunar InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 7107DED133505B88"

I tried to fix it with apt-key or gpg, but it didn't work. Is this something at your end? Thank you.

**Reply by jc00ke** [reply]
See https://github.com/orgs/regolith-linux/discussions/888

Quick fix is to reimport the key.

**Comment by kgilmer** [comment]
I will wrap up the Regolith 3.0 release on beta 4 in the next few days unless new critical issues are found

**Reply by kgilmer** [reply]
PR to cut release 3.0 tags: https://github.com/regolith-linux/voulage/pull/24

**Comment by a-shine** [comment]
Hiya,

Will there be a standalone ISO for the Regolith 3 release?

Thanks

**Reply by kgilmer** [reply]
Hi @a-shine , no ISO for 3.0.  We're almost to the Ubuntu 23.10 release.

**Reply by a-shine** [reply]
Thanks, @kgilmer - Does that mean we can expect an ISO once we hit the 23.10 release?

**Reply by kgilmer** [reply]
I'm currently working on integrating Regolith into 23.10 in it's pre-release form.  I hope to have a better understanding of how much work is required for the ISO release once that's done.  I'll reply back when more information is available.

**Comment by Sturgelose** [comment]
I'll share this here, as, at least I have the same issue. Just installed a fresh regolith and I can't share my screen at all...
https://github.com/orgs/regolith-linux/discussions/914

**Comment by kgilmer** [comment]
I'm going to lock this conversation as Regolith 3.0 shipped.
